realm/realm-library/src/main/java/io/realm/StandardRealmObjectSchema.java

```diff
@@ -676,6 +687,12 @@ private void checkFieldExists(String fieldName) {
         }
     }
 
+    private void checkAddPrimaryKeyForSync() {
```

Comment:
IMO we should strive to keep the rules for synced and non-synced Realms as close as possible. Wouldn't it make sense to disallow this for normal Realms as well?